mesonbuild/compilers/cpp.py

```diff
@@ -6,10 +6,11 @@
 import functools
 import os.path
 import typing as T
+import enum
 
 from .. import options
 from .. import mlog
-from ..mesonlib import MesonException, version_compare
+from ..mesonlib import MesonException, version_compare, File, FileOrString
```

Comment:
`FileOrString` is a just for typing, so it should go inside a `if T.TYPE_CHECKING:` block.